### PR TITLE
Update ErrorType and ErrorCode values

### PR DESCRIPTION
**Pull request by yejia-stripe**
r? @dcr-stripe 

## Summary
Update `ErrorType` values based on https://stripe.com/docs/api/errors#errors-type. Remove associated struct definitions and remove them as possible typed errors for `ResponseToError`

Update `ErrorCode` values based on https://stripe.com/docs/error-codes.

## Changelog
- ⚠️ Remove deprecated `ErrorType` values and associated struct definitions: 
  - `"api_connection_error"`, `APIConnectionError`
  - `"authentication_error"`, `AuthenticationError`
  - `"more_permissions_required"`, `PermissionError`
  - `"rate_limit_error`, `RateLimitError`
- ⚠️ Remove deprecated `ErrorCode` values:
  - `"account_already_exists"`
  - `"order_creation_failed"`
  - `"order_required_settings"`
  - `"order_status_invalid"`
  - `"order_upstream_timeout"`
  - `"upstream_order_creation_failed"`
- Add new `ErrorCode` values:

  - `"account_error_country_change_requires_additional_steps"`
  - `"account_information_mismatch"`
  - `"acss_debit_session_incomplete"`
  - `"bank_account_bad_routing_numbers"`
  - `"billing_invalid_mandate"`
  - `"cardholder_phone_number_required"`
  - `"clearing_code_unsupported"`
  - `"country_code_invalid"`
  - `"debit_not_authorized"`
  - `"instant_payouts_limit_exceeded"`
  - `"insufficient_funds"`
  - `"intent_invalid_state"`
  - `"intent_verification_method_missing"`
  - `"invoice_no_payment_method_types"`
  - `"invoice_on_behalf_of_not_editable"`
  - `"no_account"`
  - `"payment_intent_konbini_rejected_confirmation_number"`
  - `"payment_intent_mandate_invalid"`
  - `"payment_intent_payment_attempt_expired"`
  - `"payment_method_bank_account_already_verified"`
  - `"payment_method_bank_account_blocked"`
  - `"payment_method_billing_details_address_missing"`
  - `"payment_method_currency_mismatch"`
  - `"payment_method_invalid_parameter"`
  - `"payment_method_invalid_parameter_testmode"`
  - `"payment_method_microdeposit_failed"`
  - `"payment_method_microdeposit_verification_amounts_invalid"`
  - `"payment_method_microdeposit_verification_amounts_mismatch"`
  - `"payment_method_microdeposit_verification_attempts_exceeded"`
  - `"payment_method_microdeposit_verification_descriptor_code_mismatch"`
  - `"payment_method_microdeposit_verification_timeout"`
  - `"payment_method_provider_decline"`
  - `"payment_method_provider_timeout"`
  - `"payment_method_unsupported_type"`
  - `"platform_account_required"`
  - `"refer_to_customer"`
  - `"refund_disputed_payment"`
  - `"return_intent_already_processed"`
  - `"setup_intent_setup_attempt_expired"`
  - `"terminal_location_country_unsupported"`
  - `"transfer_source_balance_parameters_mismatch"`

### CODE REVIEW

**Review by dcr-stripe — COMMENTED** [review]
CI failing - I think there's a few more references to remove?

Thanks Yejia!

**Comment by yejia-stripe** [comment]
Oops! Just updated, thanks!

File: error.go [hunk 1]
```diff
@@ -29,113 +20,150 @@ type ErrorCode string
...
+	ErrorCodePaymentIntentKonbiniRejectedConfirmationNumber              ErrorCode = "payment_intent_konbini_rejected_confirmation_number"
+	ErrorCodePaymentIntentMandateInvalid                                 ErrorCode = "payment_intent_mandate_invalid"
+	ErrorCodePaymentIntentPaymentAttemptExpired                          ErrorCode = "payment_intent_payment_attempt_expired"
+	ErrorCodePaymentIntentPaymentAttemptFailed                           ErrorCode = "payment_intent_payment_attempt_failed"
+	ErrorCodePaymentIntentUnexpectedState                                ErrorCode = "payment_intent_unexpected_state"
+	ErrorCodePaymentMethodBankAccountAlreadyVerified                     ErrorCode = "payment_method_bank_account_already_verified"
+	ErrorCodePaymentMethodBankAccountBlocked                             ErrorCode = "payment_method_bank_account_blocked"
+	ErrorCodePaymentMethodBillingDetailsAddressMissing                   ErrorCode = "payment_method_billing_details_address_missing"
+	ErrorCodePaymentMethodcurrency_mismatch                              ErrorCode = "payment_method_currency_mismatch"
```
Comment (dcr-stripe) [inline error.go:113]: This and a few others didn't get capitalized correctly (linter errors).

File: stripe.go [hunk 2]
```diff
@@ -696,10 +696,6 @@ func (s *BackendImplementation) ResponseToError(res *http.Response, resBody []by
 	switch raw.Error.Type {
 	case ErrorTypeAPI:
 		typedError = &APIError{stripeErr: raw.Error}
-	case ErrorTypeAPIConnection:
-		typedError = &APIConnectionError{stripeErr: raw.Error}
```
Comment (dcr-stripe) [inline stripe.go:700]: Can we delete the underlying structs (eg. `APIConnectionError`)? Are they used by anything else?

**Review by dcr-stripe — COMMENTED** [review]
One more thing - sorry about the churn!

r? @yejia-stripe

**Pull request merged**